doozerlib/dblib.py

```diff
@@ -21,44 +148,128 @@ def __init__(self, runtime, environment, dry_run=False):
+        return int_handler
+
+    def check_table_exist(self, table_name):
+        cursor = self.connection.cursor()
+        cursor.execute("""
+            SELECT COUNT(*)
+            FROM information_schema.tables
+            WHERE table_name = '{0}'
+            """.format(table_name.replace('\'', '\'\'')))
+        if cursor.fetchone()[0] == 1:
+            cursor.close()
+            return True
+
+        cursor.close()
+        return False
+
+    def get_missing_columns(self, attr_payload, table_name):
+
+        cursor = self.connection.cursor()
+        cursor.execute(f"show columns from {table_name}")
+        existing_columns = set()
+        payload_columns = set()
+
+        for data in cursor:
+            existing_columns.add(data[0])
+
+        for column in attr_payload:
+            payload_columns.add(column)
+
+        missing_columns = list(payload_columns.difference(existing_columns))
+        return_columns = []
+        for column in missing_columns:
+            return_columns.append((column, self.column_identifier.identify(attr_payload[column])))
+        return return_columns
+
+    def handle_missing_table(self, table_name):
+
+        table_status = self.check_table_exist(table_name)
+
+        if not table_status:
+            cursor = self.connection.cursor()
+            cursor.execute(f'create table {table_name}(log_{table_name}_id bigint auto_increment primary key)')
+            cursor.close()
+
+    def handle_missing_columns(self, payload, table_name):
+
+        missing_columns = self.get_missing_columns(payload, table_name)
+
+        if missing_columns:
+            cursor = self.connection.cursor()
+
+            for column in missing_columns:
+                missing_column = column[0]
+                column_type = column[1][0]
+                cursor.execute(f"alter table {table_name} add column `{missing_column}` {column_type}")
+
+            cursor.close()
+
+    def insert_payload(self, payload, table_name):
+
+        columns, values = [], []
+
+        for column in payload:
+            columns.append(f'`{column}`')
+            values.append(f'"{payload[column]}"')
+
+        columns = ",".join(columns)
+        values = ",".join(values)
+
+        insert_string = f"insert into {table_name}({columns}) values({values})"
+
+        cursor = self.connection.cursor()
+        cursor.execute(insert_string)
+        self.connection.commit()
+        cursor.close()
+
+    @try_connecting
+    def create_payload_entry(self, payload, table_name):
+        if self.connection and self.connection.is_connected():
+            self.handle_missing_table(table_name)
+            self.handle_missing_columns(payload, table_name)
+            self.insert_payload(payload, table_name)
+            return 0
+        else:
+            return 1
+
+    @staticmethod
+    def rename_to_valid_column(column_name):
```

Comment:
I'd rather us just change the column names we use throughout the code. If I set an attribute that is invalid, throw an exception. That way, if I set an attribute, I know exactly what to include in my SQL without needing to remember this transformation.

Reply:
There is no concrete reasoning behind transforming the current column names to something. I am just trying to follow some suggested column naming conventions. So if we see the code I'm replacing . and - with _(underscore). Anything else for now passes the transformation. There is no need to remember anything. We can put any attribute name in the payload it is just replacing . and -.

Reply:
Having thought about this a little bit I have to agree with Justin. This kind of transform would only really make sense if we had a ton of legacy code that was going to need refactoring without it; we really don't. The only place there should be a column name transform is in migrating the data from simpledb to mysql... that aside, the column names used in the code should match exactly what ends up in the DB. If someone gives a bogus name (I had a look at the naming requirements, even with quoting some things are not allowed) it's probably best to detect that and fail fast (not wait for the db to give us an obtuse error).